helm/config.go

```diff
@@ -441,16 +453,23 @@ func NewEncoder(writer io.Writer, modifiers ...func(*Encoder)) *Encoder {
 		emptyLines: true,
 		indent:     2,
 		wrap:       80,
+		separator:  true,
 	}
 	enc.Set(modifiers...)
 	return enc
 }
 
-// Encode writes the config mapping held by the node to the stream.
-func (enc *Encoder) Encode(node Node) error {
+// Encode writes the config mapping held by the node to the stream. The optional
+// argument can be used to specify a prefix to indent the whole generated document.
+func (enc *Encoder) Encode(node Node, optional ...string) error {
```

Comment:
Not a fan of optional arguments like this; it's hard to figure out what optional parameters are, and doesn't work if we happen to want another non-string one.  Of course, I don't know of any good way to do that in golang either… Making the prefix non-optional might be the best we have.

Reply:
I'm fine with removing this; it is not actually being used after all.

If we ever *do* want this feature, then we could turn this into an additional option on the encoder object itself. We already have  `Indent()` for the per-level indentation, so maybe `InitialIndent()` or whatever. Not going to implement it now anyways; but I can remove the optional parameter.